src/memory.js

```diff
@@ -1,18 +1,35 @@
 class MemoryGame {
   constructor(cards) {
     this.cards = cards;
+    this.pickedCards = [];
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
     // add the rest of the class properties here
   }
 
   shuffleCards() {
     // ... write your code here
+    if (this.cards === undefined) return undefined;
+    for (let i = this.cards.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1)); // Genera un índice aleatorio entre 0 y i
+      [this.cards[i], this.cards[j]] = [this.cards[j], this.cards[i]]; // Intercambia los elementos en las posiciones i y j
+    }
+    return this.cards;
   }
 
   checkIfPair(card1, card2) {
     // ... write your code here
+    this.pairsClicked++;
+    if (card1 === card2) {
+      this.pairsGuessed++;
+      return true;
+    }
+    return false;
   }
 
   checkIfFinished() {
     // ... write your code here
+    if (this.pairsGuessed === this.cards.length / 2) return true;
+    return false;
```

Comment:
Esto esta mega perfecto!

Pero en el fondo como estas haciendo una pregunta en el if que es verdadero o falso, devuelvela sin mas!

```suggestion
    return this.pairsGuessed === this.cards.length / 2
```